accelerator/core/core.c

```diff
@@ -542,7 +543,8 @@ status_t ta_update_node_connection(ta_config_t* const ta_conf, iota_client_servi
   status_t ret = SC_OK;
   for (int i = 0; i < MAX_NODE_LIST_ELEMENTS && ta_conf->iota_host_list[i]; i++) {
     // update new IOTA full node
-    strncpy(service->http.host, ta_conf->iota_host_list[i], HOST_MAX_LEN);
+    strncpy(service->http.host, ta_conf->iota_host_list[i], HOST_MAX_LEN - 1);
+    service->http.host[HOST_MAX_LEN - 1] = 0;
```

Comment:
Replace `0` with `'\0'` since `service->http.host[HOST_MAX_LEN - 1]` is `char` data type.